app/auth/ValidatedActions.scala

```diff
@@ -0,0 +1,32 @@
+package auth
+
+import configuration.GraphQLConfiguration
+import controllers.routes
+import graphql.codegen.IsTransferAgreementComplete.{isTransferAgreementComplete => itac}
+import play.api.mvc.{Action, AnyContent, Request, Result}
+
+import scala.concurrent.ExecutionContext
+
+abstract class ValidatedActions() extends TokenSecurity {
+  implicit val ec: ExecutionContext
+  val graphqlConfiguration: GraphQLConfiguration
+
+  private val getTransferAgreementClient = graphqlConfiguration.getClient[itac.Data, itac.Variables]()
+
+  def transferAgreementExistsAction(consignmentId: Long)(f: Request[AnyContent] => Result): Action[AnyContent]
+  = secureAction.async { implicit request: Request[AnyContent] =>
+    val variables = itac.Variables(consignmentId)
+    getTransferAgreementClient.getResult(request.token.bearerAccessToken, itac.document, Some(variables)).map(data => {
```

Comment:
Should this client still be called `getTransferAgreement` now that the query name has changed?

Reply:
No, it shouldn't. I've changed it.